frontend/src/Metamaps/Topic.js

```diff
@@ -35,30 +35,14 @@ const Topic = {
   get: function (id, callback) {
     // if the desired topic is not yet in the local topic repository, fetch it
     if (Metamaps.Topics.get(id) == undefined) {
-      // console.log("Ajax call!")
-      if (!callback) {
-        var e = $.ajax({
-          url: '/topics/' + id + '.json',
-          async: false
-        })
-        Metamaps.Topics.add($.parseJSON(e.responseText))
-        return Metamaps.Topics.get(id)
-      } else {
-        return $.ajax({
-          url: '/topics/' + id + '.json',
-          success: function (data) {
-            Metamaps.Topics.add(data)
-            callback(Metamaps.Topics.get(id))
-          }
-        })
-      }
-    } else {
-      if (!callback) {
-        return Metamaps.Topics.get(id)
-      } else {
-        return callback(Metamaps.Topics.get(id))
-      }
-    }
+      $.ajax({
+        url: '/topics/' + id + '.json',
+        success: function (data) {
+          Metamaps.Topics.add(data)
+          callback(Metamaps.Topics.get(id))
+        }
+      })
+    } else callback(Metamaps.Topics.get(id))
```

Comment:
i guess we could do

`const identityFunction = x => x`
`callback = identifyFunction`